src/test/scala/edu/berkeley/ce/rockslicing/BlockVTKSpec.scala

```diff
@@ -0,0 +1,139 @@
+    listDiff
+  }
+
+  def intListElementDiff(list1: Seq[Int], list2: Seq[Int]): Seq[Int] = {
+    assert(list1.length == list2.length)
+    val listDiff  = Seq.empty[Int]
+    for (i <- 0 until list1.length) {
+      listDiff +: Seq[Int](math.abs(list1(i) - list2(i)))
+    }
+    listDiff
+  }
+
+  test("The vertices should be oriented counter-clockwise for each face") {
+    val expectedVertices = Map(
+      face1 -> face1Verts,
+      face2 -> face2Verts,
+      face3 -> face3Verts,
+      face4 -> face4Verts,
+      face5 -> face5Verts,
+      face6 -> face6Verts
+    )
+    val vtkBlock = BlockVTK(unitCube)
+    assert(expectedVertices == vtkBlock.orientedVertices)
+  }
+
+  test("List of vertices should contain only distinct vertices as tuples") {
+    val vtkBlock = BlockVTK(unitCube)
+    val expectedVertices = List((0.0, 1.0, 1.0), (0.0, 0.0, 1.0), (1.0, 0.0, 1.0), (1.0, 1.0, 1.0),
+                                (0.0, 1.0, 0.0), (0.0, 0.0, 0.0), (1.0, 0.0, 0.0), (1.0, 1.0, 0.0))
+    assert(expectedVertices == vtkBlock.tupleVertices)
+  }
+
+  test("List of tuple vertices should flatten into list of doubles") {
+    val vtkBlock = BlockVTK(unitCube)
+    val expectedVertices = List((0.0, 1.0, 1.0), (0.0, 0.0, 1.0), (1.0, 0.0, 1.0), (1.0, 1.0, 1.0),
+                                (0.0, 1.0, 0.0), (0.0, 0.0, 0.0), (1.0, 0.0, 0.0), (1.0, 1.0, 0.0)) flatMap { t =>
+      List(t._1, t._2, t._3)
+    }
+    val verticesDiff = doubleListElementDiff(expectedVertices, vtkBlock.vertices)
+    assert(verticesDiff.filter(_ > NumericUtils.EPSILON).isEmpty)
+  }
+
+  test("Vertex ID's should start from 0 and go up to 7") {
+    val vtkBlock = BlockVTK(unitCube)
+    val expectedIDs = List((0.0, 1.0, 1.0), (0.0, 0.0, 1.0), (1.0, 0.0, 1.0), (1.0, 1.0, 1.0),
+                           (0.0, 1.0, 0.0), (0.0, 0.0, 0.0), (1.0, 0.0, 0.0), (1.0, 1.0, 0.0)).indices
+    val idsDiff = intListElementDiff(expectedIDs.toSeq, vtkBlock.vertexIDs.toSeq)
+    assert(idsDiff.filter(_ > NumericUtils.EPSILON).isEmpty)
+  }
+
+  test("Connectivities for each face should be in terms of vertex ID's") {
+    val vtkBlock = BlockVTK(unitCube)
+    val faceConnectivities = Map(
+      face1 -> List[Int](4, 5, 1, 0),
+      face2 -> List[Int](3, 2, 6, 7),
+      face3 -> List[Int](1, 5, 6, 2),
+      face4 -> List[Int](4, 0, 3, 7),
+      face5 -> List[Int](7, 6, 5, 4),
+      face6 -> List[Int](0, 1, 2, 3)
+    )
+    assert(faceConnectivities == vtkBlock.connectivityMap)
+  }
+
+  test("Connectivity map should flatten into list of integers") {
+    val vtkBlock = BlockVTK(unitCube)
+    val faceConnectivities = Map(
+      face1 -> List[Int](4, 5, 1, 0),
+      face2 -> List[Int](3, 2, 6, 7),
+      face3 -> List[Int](1, 5, 6, 2),
+      face4 -> List[Int](4, 0, 3, 7),
+      face5 -> List[Int](7, 6, 5, 4),
+      face6 -> List[Int](0, 1, 2, 3)
+    ).values.flatten
+    val connectionsDiff = intListElementDiff(faceConnectivities.toSeq, vtkBlock.connectivity.toSeq)
```

Comment:
I could be wrong, but I think the explicit conversion is required here (at least that's what the compiler says). The inputs are initially `Iterable[Int]` and the code won't compile until they are explicitly converted to `Seq[Int]`, so perhaps there's something I'm missing?